### PR TITLE
Feature/blend mode reverse subtract

### PR DESCRIPTION
**Pull request by Krozark**
see issue #944

### CODE REVIEW

**Comment by LaurentGomila** [comment]
Since they are not used often, I don't think there should be a corresponding preset for them.

File: src/SFML/Graphics/GLExtensions.hpp [hunk 1]
```diff
@@ -64,6 +64,9 @@
     #define GLEXT_glBlendEquation                     glBlendEquationOES
     #define GLEXT_GL_FUNC_ADD                         GL_FUNC_ADD_OES
     #define GLEXT_GL_FUNC_SUBTRACT                    GL_FUNC_SUBTRACT_OES
+	#define GLEXT_GL_FUNC_REVERSE_SUBTRACT GL_FUNC_REVERSE_SUBTRACT_OES
```
Comment (TankOs) [inline src/SFML/Graphics/GLExtensions.hpp]: Please use 4 spaces for each indentation level, not tabs.

**Comment by TankOs** [comment]
I think presets should exist for all or none.

**Comment by Krozark** [comment]
Sorry for the tabs :/ Usually I use 4 space. I will fix this.

What about the preset? The default have to exist (used as default parameter), but what about the others?

**Comment by eXpl0it3r** [comment]
I'm with @LaurentGomila here. We provide the possibility to use all the OpenGL blend modes, but only expose the common ones as preset.

As for the code, also make sure that the comments actually line up.

**Comment by LaurentGomila** [comment]
There are too many combinations of equation, source mode, destination mode, source alpha mode and destination alpha mode... We can't have a preset for all of them ;)

And the current presets are for common combinations, not just common equations (look at Multiply).

**Comment by Krozark** [comment]
Ok. So I've remove them.

**Comment by eXpl0it3r** [comment]
Space indenting is still not corrected everywhere and the comments are not aligned.
Additionally you should squash the commits into one.

**Comment by Krozark** [comment]
@eXpl0it3r "Additionally you should squash the commits into one".

How do you do that?

**Comment by Krozark** [comment]
All is done (I hope).

**Comment by kimci86** [comment]
There are still tabs before the `#define` lines. Is the blank line in GLExtensions.hpp intended (line 149) ? Moreover I guess the commit message is not explanatory enough.

**Comment by Krozark** [comment]
@kimci86  Thanks.

**Comment by Krozark** [comment]
@kimci86  Thanks.

**Comment by Bromeon** [comment]
The [original forum discussion](http://en.sfml-dev.org/forums/index.php?topic=18797.0) is still running, let's not split it to three places (forum, issue, pull request). I don't think a link to the forum needs to be part of the commit message.

This pull request merge-conflicts with https://github.com/SFML/SFML/pull/938, we should wait to change the used OpenGL extensions before we have met a decision there.

**Comment by TankOs** [comment]
The forum discussion hasn't had replies for 3 days, so at least all (active) team members have read it and did not reply. IMHO the census is that this is a good idea. We've even been working on getting this merge-ready, and I don't know why we should discuss this any further. 

Please let's not slow down progress due to already happened discussions or other pull requests that are waiting for something, either.

Also this is a simple, straightforward and non-API-breaking change.

**Comment by Bromeon** [comment]
> The forum discussion hasn't had replies for 3 days

Which three days are those? There were replies on August 13, 14, 15, 16, 17. Additionally I wasn't there for some days; we have the AFK thread for a reason.

Furthermore, Laurent and Krozark think we can add `ReverseSubtract` but not `Max` and `Min`, so I wouldn't say everybody agrees.

**Comment by Krozark** [comment]
If all is fine with this version, I will squash all commit together with a better message.

File: include/SFML/Graphics/BlendMode.hpp [hunk 2]
```diff
@@ -68,8 +68,9 @@ struct SFML_GRAPHICS_API BlendMode
     ////////////////////////////////////////////////////////
     enum Equation
     {
-        Add,     ///< Pixel = Src * SrcFactor + Dst * DstFactor
-        Subtract ///< Pixel = Src * SrcFactor - Dst * DstFactor
+        Add,             ///< Pixel = Src * SrcFactor + Dst * DstFactor
+        Subtract,        ///< Pixel = Src * SrcFactor - Dst * DstFactor
+        ReverseSubtract ///< Pixel = Dst * DstFactor - Src * SrcFactor
```
Comment (kimci86) [inline include/SFML/Graphics/BlendMode.hpp]: The comments are not indented properly.

**Comment by TankOs** [comment]
> Which three days are those?

I was on the wrong page, sorry.

> so I wouldn't say everybody agrees.

Okay. My opinion is well-known anyway, so feel free to discuss this further.

**Comment by Bromeon** [comment]
Looks good, you can squash the commits in my opinion.

By the way, is it just me or where is the automated build result of this PR?

**Comment by TankOs** [comment]
@Bromeon 
They only trigger for feature and bugfix branches, unfortunately. I still have to setup the Buildbot instance. :runner:

**Comment by Bromeon** [comment]
:+1:

**Comment by eXpl0it3r** [comment]
This PR has been added to my merge list, meaning it will be merged soon, unless someone raises any concerns.

**Comment by eXpl0it3r** [comment]
Merged in 7ff9478061e369d31228c2f98f8ec91f02de49d5

**Pull request closed without merging**